TaskWebApp/App_Start/Startup.Auth.cs

```diff
@@ -141,18 +157,38 @@ private async Task OnAuthorizationCodeReceived(AuthorizationCodeReceivedNotifica
             // Extract the code from the response notification
             var code = notification.Code;
 
-            string signedInUserID = notification.AuthenticationTicket.Identity.FindFirst(ClaimTypes.NameIdentifier).Value;
-            TokenCache userTokenCache = new MSALSessionCache(signedInUserID, notification.OwinContext.Environment["System.Web.HttpContextBase"] as HttpContextBase).GetMsalCacheInstance();
-            ConfidentialClientApplication cca = new ConfidentialClientApplication(ClientId, Authority, RedirectUri, new ClientCredential(ClientSecret), userTokenCache, null);
+            string signedInUserID = notification.JwtSecurityToken.Subject; //notification.AuthenticationTicket.Identity.FindFirst(Startup.ClaimsSubject).Value;
+            ConfidentialClientApplication cca = GetConfidential();
+            var httpContextBase = notification.OwinContext.Environment["System.Web.HttpContextBase"] as HttpContextBase;
+            HttpContext httpContext =  httpContextBase.ApplicationInstance.Context;
+            TokenCacheHelper.EnablePersistence(cca.UserTokenCache);
```

Comment:
You should not use a file based token cache on a Web app:
1) Web apps usually don't have access to the disk
2) Web apps should have one cache per user (here the code you used is for public client applications)

@bgavrilMS @henrik-me @MarkZuber @kalyankrishna1 can you please follow-up with @valnav while I'm OOF.

Reply:
I'm looking